addons/urdfreader/urdfreader.cc

```diff
@@ -59,6 +59,197 @@ namespace RigidBodyDynamics
 
 // =============================================================================
 
+Joint get_rbdl_joint(const JointPtr &urdf_joint)
+{
+  Joint rbdl_joint;
+  if (urdf_joint->type == UrdfJointType::REVOLUTE ||
+      urdf_joint->type == UrdfJointType::CONTINUOUS) {
+    rbdl_joint = Joint(SpatialVector(urdf_joint->axis.x, urdf_joint->axis.y,
+                                     urdf_joint->axis.z, 0., 0., 0.));
+  } else if (urdf_joint->type == UrdfJointType::PRISMATIC) {
+    rbdl_joint = Joint(SpatialVector(0., 0., 0., urdf_joint->axis.x,
+                                     urdf_joint->axis.y, urdf_joint->axis.z));
+  } else if (urdf_joint->type == UrdfJointType::FIXED) {
+    rbdl_joint = Joint(JointTypeFixed);
+  } else if (urdf_joint->type == UrdfJointType::FLOATING) {
+    // todo: what order of DoF should be used?
+    rbdl_joint = Joint(
+      SpatialVector(0., 0., 0., 1., 0., 0.),
+      SpatialVector(0., 0., 0., 0., 1., 0.),
+      SpatialVector(0., 0., 0., 0., 0., 1.),
+      SpatialVector(1., 0., 0., 0., 0., 0.),
+      SpatialVector(0., 1., 0., 0., 0., 0.),
+      SpatialVector(0., 0., 1., 0., 0., 0.));
+  } else if (urdf_joint->type == UrdfJointType::PLANAR) {
+    // todo: which two directions should be used that are perpendicular
+    // to the specified axis?
+    ostringstream error_msg;
+    error_msg << "Error while processing joint '" << urdf_joint->name
+              << "': planar joints not yet supported!" << endl;
+    throw RBDLFileParseError(error_msg.str());
+  }
+  return rbdl_joint;
+}
+
+Body get_rbdl_body(const LinkPtr &urdf_link, bool is_root_link)
+{
+  // assemble the body
+  urdf::Vector3 link_inertial_rpy_temp;
+  Vector3d link_inertial_position;
+  Vector3d link_inertial_rpy;
+  Matrix3d link_inertial_inertia = Matrix3d::Zero();
+  double link_inertial_mass = 0.;
+
+
+  // but only if we actually have inertial data
+#ifdef RBDL_USE_ROS_URDF_LIBRARY
+  if (urdf_link->inertial) {
+    auto I = urdf_child->inertial;
+#else
+  if (urdf_link->inertial.has_value()) {
+    auto I = &urdf_link->inertial.value();
+#endif
+    link_inertial_mass = I->mass;
+
+    link_inertial_position.set(
+      I->origin.position.x,
+      I->origin.position.y,
+      I->origin.position.z);
+
+    if(!is_root_link)
+    {
+      I->origin.rotation.RPY(link_inertial_rpy_temp.x,
+                                 link_inertial_rpy_temp.y,
+                                 link_inertial_rpy_temp.z);
+      link_inertial_rpy.set(link_inertial_rpy_temp.x, link_inertial_rpy_temp.y, link_inertial_rpy_temp.z);
```

Comment:
this section seems like it would need some alignment and also check that the 80char line limit is not reached.

(I know not all code in the addons has had the proper review of the style 😅, so this might just be left overs from before)

Reply:
Yes, it was the hardest part, as the styling guide that you specified to me seems to be not applied for this addons, as when I tried to apply them, there were lot of formatting changes. I didn't want to have commit them, so instead I did apply them manually to my part of changes

Reply:
Done